pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/DataTypeTransformer.java

```diff
@@ -95,7 +103,12 @@ public GenericRow transform(GenericRow record) {
 
         record.putValue(column, value);
       } catch (Exception e) {
-        throw new RuntimeException("Caught exception while transforming data type for column: " + column, e);
+        if (!_useDefaultValueOnError) {
+          throw new RuntimeException("Caught exception while transforming data type for column: " + column, e);
+        } else {
+          LOGGER.debug("Caught exception while transforming data type for column: {}", column, e);
+          record.putValue(column, null);
```

Comment:
is it intended to always set null here and let NullValueTransformer (which happens after DataTypeTransformer) to change null to the default value set for nullValueHandling?It's pretty neat to me.

Reply:
Yes. Since `NullValueTransformer` takes care of this, I don't want to duplicate that logic and introduce the risk of it diverging from the current logic in future patches.